docs/tutorials/HowToBuildADriver.md

```diff
@@ -0,0 +1,162 @@
+using Moryx.Serialization;
+
+namespace Moryx.Resources.Samples.DriverTutorial
+{
+    [ResourceRegistration]
+    [DisplayName("Example Driver"), Description("An example driver")]
+    public class ExampleDriver : Driver, IExampleDriver
+    {
+        [DataMember, EntrySerialize]
+        public string AStringValue { get; set; }
+
+        [DataMember, EntrySerialize]
+        public int AnIntValue { get; set; }
+
+        public string ANonEntrySerializeMember { get; set; }
+
+        [EntrySerialize, DisplayName("Square"), Description("Just multiplies given value with itself")]
+        public int Square(int value)
+        {
+            return value * value;
+        }
+    }
+}
+````
+
+The implementation of the `ExampleDriver` derives from the [Driver](../../src/Moryx.AbstractionLayer/Drivers/Driver.cs) base class. It also implements the `IDriver` interface. The next important thing is the [ResourceRegistration attribute](../../src/Moryx.AbstractionLayer/Resources/Attributes/ResourceRegistrationAttribute.csResourceRegistrationAttribute). The AbstractionLayer can now identify this driver as a resource. Additional attributes like `DisplayName` and `Description` are used within the Resource UI.
+
+The two properties `AStringValue` and `AnIntValue` are shown in the ResourceUI and can be edited by the user. The member `ANonEntrySerializeMember` is invisible for the user and is only used inside the AbstractionLayer.
+The `Square` function is also visible in the Resource UI. And: It is callable from there.
+
+## Lifecycle and StateMachine
+
+The `ExampleDriver` is just a simple implementation for a driver. As like every [Resource](../../src/Moryx.AbstractionLayer/Resources/Resource.cs) you can `Initialize`, `Start`, `Stop` a driver. Also `State machine` support is built in:
+
+````cs
+using Moryx.AbstractionLayer.Drivers;
+using Moryx.AbstractionLayer.Resources;
+
+namespace Moryx.Resources.Samples.DriverTutorial
+{
+    
+    [ResourceRegistration]
+    [DisplayName("StateExample Driver"), Description("An example driver that uses the state machine")]
+    public class StateExampleDriver : Driver, IExampleDriver, IStateContext
+    {
+        ...
+
+        private ExampleStateBase _state;
+
+        private readonly object _stateLock = new object();
+
+        /// <seealso cref="IDriver"/>
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            StateMachine.Initialize<ExampleStateBase>(this);
+        }
+
+        ...
+
+        public override void OnStart(){
+            lock(_stateLock)
+                State.Connect();
+        }
+    }
+
+    
+}
+````
+
+In order to create the State machine, first define a base type, the methods you need and your states.  States are derived from `StateBase`. For drivers there exists a specific `DriverState`, which includes the methods `Connect()` and `Disconnect()`. Typical States for a Driver are `Disconnected`, `Connecting` and `Connected`.
+
+```C#
```

Comment:
If you use cs instead of c# github recognizes the C# code formatting also in the git compare 🤔